client/src/controllers/DropdownController.ts

```diff
@@ -113,20 +95,22 @@ export class DropdownController extends Controller<HTMLElement> {
   static targets = ['toggle', 'content'];
   static values = {
     hideOnClick: { default: false, type: Boolean },
+    keepMounted: { default: false, type: Boolean },
     offset: Array,
     theme: { default: 'dropdown' as TippyTheme, type: String },
   };
 
   // Hide on click *inside* the dropdown. Differs from tippy's hideOnClick
   // option for outside clicks that defaults to true and we don't yet expose it.
-  declare hideOnClickValue: boolean;
-  declare offsetValue: [number, number];
+  declare readonly hideOnClickValue: boolean;
+  declare readonly keepMountedValue: boolean;
+  declare readonly offsetValue: [number, number];
+  declare readonly hasOffsetValue: boolean;
+  declare readonly themeValue: TippyTheme;
```

Comment:
I have leaned towards making all values not read only in Stimulus code as they actually can be changed.

However, it could be a bit nuanced as we may not really want the controller to change these values (e.g. we may not have code to support he theme value changing after connect).

I am not sure what the right way is, it's probably not critical but just adding a comment.

Reply:
I'd personally prefer to use `readonly` while we don't have any code that changes the value, and remove the `readonly` when we do. We can think of this as somewhat similar to `const` vs `let`.

Also, as far as I'm aware this is only enforced via TypeScript and not on runtime, so it doesn't really restrict the behaviour. However, I think it helps developers to carefully consider *why* a value needs to be changed and what the effect should be (which means you'll need to write the value changed hook). That's just a matter of preference though, but if we decide to remove `readonly` then I'd prefer us doing it everywhere (except Stimulus' default ones like `hasFooValue`).

Reply:
Yeah. I agree with this, I don't think it needs changing but more was wondering what we should be doing generally in other code. 

Thanks Sage. That's a good point about the TS enforcement only also.